src/main/kotlin/io/ionic/libs/iongeolocationlib/controller/IONGLOCController.kt

```diff
@@ -51,12 +76,17 @@ class IONGLOCController(
         try {
             val checkResult: Result<Unit> =
                 checkLocationPreconditions(activity, options, isSingleLocationRequest = true)
-            return if (checkResult.isFailure) {
+            return if (checkResult.shouldNotProceed(options)) {
```

Comment:
Minor: we could do return try and simplify the code. But it's more a matter of preference really.